### DISCUSSION TITLE
How to debug item properties in runtime?

### DISCUSSION THREAD

**Discussion by qarmin** [slint-ui/slint#3727] [Q&A] [answered]
In QML I'm able to print from qml value of any item, e.g.
```
onWidthChanged: {
    console.error(width)
}
```
is this possible to do from slint file?
Or maybe is there way to print values of properties of every item with different method? Something like gnome inspector? - https://wiki.gnome.org/Projects/GTK/Inspector

**Comment by ogoffart** [comment, marked as answer]
There is a `debug (...)` that works the same way:
https://slint.dev/releases/1.2.2/docs/slint/src/language/builtins/functions#debug
You can just use it in the binding itself

**Reply by qarmin** [reply]
Can this be added to https://slint.dev/releases/1.2.2/docs/slint/src/advanced/debugging_techniques ?

This was the place where I expected to found it and where I started to search

**Reply by hunger** [reply]
Good point, that should really be mentioned there. Is #3731 helpful enough?
